test/server/core/gameSteps/phases/SetupPhase.spec.ts

```diff
@@ -0,0 +1,100 @@
+describe('Setup Phase', function() {
+    integration(function(contextRef) {
+        describe('Setup Phase', function() {
+            beforeEach(function () {
+                contextRef.setupTest({
+                    phase: 'setup',
+                    player1: {
+                        resources: [],
+                        deck: ['armed-to-the-teeth',
+                            'collections-starhopper',
+                            'covert-strength',
+                            'chewbacca#pykesbane',
+                            'battlefield-marine',
+                            'battlefield-marine',
+                            'battlefield-marine',
+                            'moment-of-peace',
+                            'moment-of-peace',
+                            'moment-of-peace',
+                            'moment-of-peace',
+                        ],
+                    },
+                    player2: {
+                        resources: [],
+                        deck: [
+                            'moisture-farmer',
+                            'atst',
+                            'atst',
+                            'atst',
+                            'atst',
+                            'atst',
+                            'atst',
+                            'atst',
+                            'wampa',
+                            'atst',
+                            'atst',
+                        ],
+                    }
+                });
+            });
+
+            it('should flow normally through each step', function () {
+                const { context } = contextRef;
+
+                // The setup phase is divided into 3 or 4 steps: choose Initiative player, Draw card step, mulligan step (optional), resource step
+                // Choose Initiative Step
+                context.selectInitiativePlayer(context.player1);
+
+                // Draw cards step
+                expect(context.player1.handSize).toBe(6);
+                expect(context.player2.handSize).toBe(6);
+                const beforeMulliganHand = context.player1.hand;
+
+                // Mulligan step
+                context.player1.clickPrompt('yes');
+                context.player2.clickPrompt('no');
+                const afterMulliganHand = context.player1.hand;
+                expect(beforeMulliganHand).not.toEqual(afterMulliganHand);
```

Comment:
Also need to check that player2's hand has not changed

Reply:
gotcha!